WarehouseManagement.Core/Services/MarkerService.cs

```diff
@@ -0,0 +1,213 @@
+﻿using Microsoft.EntityFrameworkCore;
+using WarehouseManagement.Core.Contracts;
+using WarehouseManagement.Core.DTOs.Marker;
+using WarehouseManagement.Infrastructure.Data.Common;
+using WarehouseManagement.Infrastructure.Data.Models;
+
+namespace WarehouseManagement.Core.Services;
+
+public class MarkerService : IMarkerService
+{
+    private readonly IRepository repository;
+
+    public MarkerService(IRepository repository)
+    {
+        this.repository = repository;
+    }
+
+    public async Task AddAsync(MarkerFormDto model, string userId)
+    {
+        if (model != null)
+        {
+            var marker = new Marker
+            {
+                Name = model.Name,
+                CreatedAt = DateTime.UtcNow,
+                CreatedByUserId = userId,
+            };
+            await repository.AddAsync(marker);
+            await repository.SaveChangesAsync();
+        }
+    }
+
+    public async Task DeleteAsync(int id, string userId)
+    {
+        var marker = await repository.GetByIdAsync<Marker>(id);
+        if (marker == null)
+        {
+            throw new KeyNotFoundException($"Marker with ID {id} not found.");
+        }
+
+        marker.IsDeleted = true;
+        marker.DeletedAt = DateTime.UtcNow;
+        marker.DeletedByUserId = userId;
+
+        repository.SoftDelete(marker);
+        await repository.SaveChangesAsync();
+    }
+
+    public async Task EditAsync(int id, MarkerFormDto model, string userId)
+    {
+        var marker = await repository.GetByIdAsync<Marker>(id);
+        if (marker == null)
+        {
+            throw new KeyNotFoundException($"Marker with ID {id} not found.");
+        }
+
+        marker.Name = model.Name;
+        marker.LastModifiedAt = DateTime.UtcNow;
+        marker.LastModifiedByUserId = userId;
+
+        await repository.SaveChangesWithLogAsync();
+    }
+
+    public async Task<bool> ExistByNameAsync(string name)
+    {
+        return await repository
+            .AllReadOnly<Marker>()
+            .AnyAsync(m => m.Name.ToLower() == name.ToLower());
+    }
+
+    public async Task<IEnumerable<MarkerDto>> GetAllAsync()
```

Comment:
keep in mind at some point we will have pagination/filters etc for GET endpoints. For now i guess we can just use this as it is, but I will make sure to create a story so you guys can take a look at it and see if it will be some Base Class or whatever, you know more than me for this so you will figure it out😄

Reply:
I will start implementing it now on Marker/all endpoint.

Reply:
Lets have a different branch for this.